app/scripts/controllers/landing.js

```diff
@@ -28,26 +28,28 @@ angular.module('confRegistrationWebApp')
     ];
     $scope.jumboImg = jumboImgs[_.random(0, jumboImgs.length - 1)];
 
-    $scope.eventSearch = _.throttle(function(val) {
+    $scope.eventSearch = _.debounce(function(val) {
       if(!val){ return; }
       if(val.length < 2){
         return;
       }
-      $http.get('conferences', {
-        params: {
-          conferenceName: val
-        }
-      }).then(function(response){
-        $scope.eventFilters = {
-          locationName: '',
-          date: ''
-        };
-        $scope.searchComplete = val;
-        $scope.searchResults = response.data;
+      if(angular.isDefined($scope.searchVal) && val === $scope.searchVal) {
```

Comment:
I tested both ways and seemed to operate the same whether I placed around the HttpPromise or within '''then'''.  I placed it within '''then'''.